src/psyclone/tests/psyir/nodes/type_convert_intrinsic_test.py

```diff
@@ -60,44 +60,47 @@ def test_type_convert_binaryop_create(operation, op_str):
     wp_sym = DataSymbol("wp", INTEGER_SINGLE_TYPE)
     # Reference to a kind parameter
     rhs = Reference(wp_sym)
-    binaryoperation = BinaryOperation.create(operation, lhs, rhs)
-    assert binaryoperation._operator is operation
-    check_links(binaryoperation, [lhs, rhs])
-    result = FortranWriter().binaryoperation_node(binaryoperation)
-    assert op_str + "(tmp1, wp)" in result.lower()
+    intr_call = IntrinsicCall.create(intrinsic, [lhs, ("kind", rhs)])
+    assert intr_call.intrinsic is intrinsic
+    check_links(intr_call, [lhs, rhs])
+    result = fortran_writer(intr_call)
+    assert intr_str + "(tmp1, kind=wp)" in result.lower()
     # Kind specified with an integer literal
     rhs = Literal("4", INTEGER_SINGLE_TYPE)
-    binaryoperation = BinaryOperation.create(operation, lhs.detach(), rhs)
-    check_links(binaryoperation, [lhs, rhs])
-    result = FortranWriter().binaryoperation_node(binaryoperation)
-    assert op_str + "(tmp1, 4)" in result.lower()
+    intr_call = IntrinsicCall.create(intrinsic, [lhs.detach(), ("kind", rhs)])
+    check_links(intr_call, [lhs, rhs])
+    result = fortran_writer(intr_call)
+    assert intr_str + "(tmp1, kind=4)" in result.lower()
     # Kind specified as an arithmetic expression
     rhs = BinaryOperation.create(BinaryOperation.Operator.ADD,
                                  Reference(wp_sym),
                                  Literal("2", INTEGER_SINGLE_TYPE))
-    binaryoperation = BinaryOperation.create(operation, lhs.detach(), rhs)
-    check_links(binaryoperation, [lhs, rhs])
-    result = FortranWriter().binaryoperation_node(binaryoperation)
-    assert op_str + "(tmp1, wp + 2)" in result.lower()
+    intr_call = IntrinsicCall.create(intrinsic, [lhs.detach(), ("kind", rhs)])
+    check_links(intr_call, [lhs, rhs])
+    result = fortran_writer(intr_call)
+    assert intr_str + "(tmp1, kind=wp + 2)" in result.lower()
 
 
 @pytest.mark.xfail(reason="Only limited checking is performed on the "
```

Comment:
The 'reason' text needs updating. Also, is it possible to move the xfail down into the test , e.g.:

        if something_wrong:
            pytest.xfail("#641 Loop variables are stored.")

Reply:
I have updated the reason but I am unsure if this is what you meant.

Also the issue is that pytest.raises are not triggered, so I am unsure how to bring the xfail inline without fundamentally refactoring the test, that will need to be reverted when the issue is resolved.